### PR TITLE
Handle therapy bookings webhooks

### PR DESCRIPTION
**Pull request by Seyi-Baks**
[Ticket](https://www.notion.so/chayn/Handle-therapy-bookings-webhooks-065e6ad67b8f45aea4645a6b9e8811a6)

This PR contains:
- Zapier Webhook endpoint 

## Endpoint 
- Zapier Webhook endpoint - `POST api/v1/partner-access/webhooks/simplybook`

Use the calendar integrations webhooks (zapier) to listen for booking/cancellation events and make these updates to the PartnerAccess model.

### CODE REVIEW

File: src/utils/constants.ts [hunk 1]
```diff
@@ -4,6 +4,11 @@ import { UserEntity } from '../entities/user.entity';
 export const isProduction = process.env.NODE_ENV === 'production';
 export const isDevelopment = process.env.NODE_ENV === 'development';
 
+export enum ZAPIER_ACTION_ENUM {
+  NEW_BOOKING = 'NEW_BOOKING',
+  CANCELED_BOOKING = 'CANCELED_BOOKING',
```
Comment (annarhughes) [inline src/utils/constants.ts:9]: typo canceled -> cancelled

File: src/partner-access/zapier-auth.guard.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+    const zapierToken = authorization.split('Basic ')[1];
+
+    if (zapierToken !== process.env.ZAPIER_TOKEN) {
+      throw new UnauthorizedException('Unauthorized: invalid auth token');
+    }
+
+    return true;
+  }
+}
```
Comment (annarhughes) [inline src/partner-access/zapier-auth.guard.ts:22]: we set up a username/password on the zapier webhooks, shall we include a check here for those? I guess storing the username/password as env variables?

**Review by annarhughes — COMMENTED** [review]
Nice work, v clean! 
I'd suggest renaming the `ZapierBodyDto` and `ZAPIER_ACTION_ENUM` to `SimplybookBodyDto and `SIMPLYBOOK_ACTION_ENUM` just because its more verbose and allows us to use zapier for something else later. Couple more minor comments

File: src/partner-access/zapier-auth.guard.ts (hunk 2 above)
Comment (Seyi-Baks) [reply src/partner-access/zapier-auth.guard.ts:22]: The username and password aren't sent back in the request header or in the body of the object.

**Pull request merged**